### PR TITLE
UI changes and some core feature added

### PR DESCRIPTION
**Pull request by cryogon**
### Changelog 
**UI**
1. Added Option Menu where we can create the new path and get the list of path
2. Navbar now shows the username of person who is logged in

**CORE**
1. User can now login and signup
2. User will get their path list
1. User can now draw line and it will be updated on real-time
3. User can now store the path in database and retrieve it as a list

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1 @@
+process.env.VUE_APP_API_TARGET = "https://dashing-mouse-23.hasura.app/v1/graphql";
```
Comment (blahmonkey) [inline .env]: @JatinThakur14 You dont need to write `process.env.`

Should just be `VUE_APP_API_TARGET=xxx`

File: src/components/LoginForm.vue [hunk 2]
```diff
@@ -1,38 +1,162 @@
 <template>
   <div class="card">
     <div class="header">
-      <h1>LOGIN</h1>
+      <h1 @click="inLoginForm = true">LOGIN</h1>
+      <h1 @click="inLoginForm = false">Signup</h1>
```
Comment (blahmonkey) [inline src/components/LoginForm.vue]: @JatinThakur14 Make both the same kind of case i.e. uppercase or capitalized

File: src/components/LoginForm.vue [hunk 3]
```diff
@@ -1,38 +1,162 @@
...
       <input
         type="password"
         name="password"
         id="passwordInput"
         class="inputField"
         placeholder="Password"
+        v-model="password"
       />
       <label for="password" class="inputLabel" id="passwordLabel"
         >Password
```
Comment (blahmonkey) [inline src/components/LoginForm.vue]: @JatinThakur14 Same here. username/password should have same casing

File: src/components/LoginForm.vue [hunk 4]
```diff
@@ -1,38 +1,162 @@
...
+      <input
+        type="password"
+        name="password"
+        id="passwordInput"
+        class="inputFieldSignup"
+        placeholder="Password"
+        v-model="password"
+      />
+      <label for="password" class="inputLabelSignup" id="passwordLabel"
+        >Password
```
Comment (blahmonkey) [inline src/components/LoginForm.vue]: @JatinThakur14 same here with casing

File: src/components/LoginForm.vue [hunk 5]
```diff
@@ -1,38 +1,162 @@
...
+      </label>
+      <button @click.prevent="doSignup">Login</button>
```
Comment (blahmonkey) [inline src/components/LoginForm.vue]: @JatinThakur14 This should be signup not login

File: src/components/LoginForm.vue [hunk 6]
```diff
@@ -1,38 +1,162 @@
...
+      password: "",
+      email: "",
+      inLoginForm: true,
+    };
+  },
   methods: {
-    doLogin() {
-      this.$store.dispatch("login");
+    async doLogin() {
+      localStorage.setItem("token", this.username);
```
Comment (blahmonkey) [inline src/components/LoginForm.vue:71]: @JatinThakur14 Why do you need this line now

File: src/components/LoginForm.vue [hunk 7]
```diff
@@ -1,38 +1,162 @@
...
+              password
+            }
+          }
+        `,
+      });
+      console.log(data.data.users[0].username);
+      try {
+        if (
+          data.data.users[0].username === this.username &&
+          data.data.users[0].password == this.password
```
Comment (blahmonkey) [inline src/components/LoginForm.vue:85]: @JatinThakur14 What is this logic for

File: src/components/LoginForm.vue [hunk 8]
```diff
@@ -1,38 +1,162 @@
...
+        ) {
+          this.$store.dispatch("login");
+        }
+      } catch (err) {
+        alert("Email or Password is wrong");
+      }
     },
+    async doSignup() {
+      if (!this.name && !this.email && !this.username && !this.password) {
```
Comment (blahmonkey) [inline src/components/LoginForm.vue]: @JatinThakur14 I dont know why you're using email and username. Just use one. And you dont need name

File: src/components/MapComponent.vue [hunk 9]
```diff
@@ -0,0 +1,257 @@
...
+
+      <div class="listOfPaths">
+        <ol>
+          <li
+            v-for="path in paths"
+            :key="path.name"
+            class="path"
+            @click="
+              () => {
+                this.locations = path.path.path;
```
Comment (blahmonkey) [inline src/components/MapComponent.vue]: @JatinThakur14 path.path.path is absolutely terrible object naming

**Review by blahmonkey — CHANGES_REQUESTED** [review]

File: src/components/LoginForm.vue (hunk 6 above)
Comment (cryogon) [reply src/components/LoginForm.vue:71]: Its for setting a temporary token so the graphql query can be send by that user and if the query found result than we can check the username and password to permit the login access

File: src/components/LoginForm.vue (hunk 7 above)
Comment (cryogon) [reply src/components/LoginForm.vue:85]: It for checking that the given username and password is available in database or not

File: src/components/LoginForm.vue (hunk 8 above)
Comment (cryogon) [reply src/components/LoginForm.vue]: I will keep that in mind and will fix this in next patch

File: src/components/MapComponent.vue (hunk 9 above)
Comment (cryogon) [reply src/components/MapComponent.vue]: I will fix this in next patch. I was just trying to if its working or not

**Review by blahmonkey — APPROVED** [review]

**Pull request merged**